### DISCUSSION TITLE
make own smplayer for mac

### DISCUSSION THREAD

**Discussion by tonho911** [smplayer-dev/smplayer#874] [Q&A] [answered]
where can I find the method for Mac, because I'm curious what it would be like to build it myself
thank you smplayer dev if you are willing to teach me

**Comment by smplayer-dev** [comment, marked as answer]
This is the github action that builds smplayer for Mac, you can see there the commands to build the package:
https://github.com/smplayer-dev/smplayer/blob/master/.github/workflows/build-macos.yml

**Reply by tonho911** [reply]
this version
mpv-url: https://github.com/smplayer-dev/mpv/releases/download/v0.36.0/mpv-macos-10.15-av.tar.bz2
can I replace this version with the publicly released version (v.0.37) which is here?
https://laboratory.stolendata.net/~djinn/mpv_osx/

**Comment by smplayer-dev** [comment]
It will probably work but the video will be on a separate window. I'm using a patched version to fix that problem.

**Reply by tonho911** [reply]
no problem the video will be played in a different window bro. My real problem is when playing offline videos that I download from here https://www.youtube.com/watch?v=kF-0q042Jjk
 
and several others (offline by disk) all the colors produced are very different from the original and don't look natural anymore when played on a Mac but that's not the case when played on Windows 10, the colors match and I even like them because they are really similar with potplayer.
The strange thing that I felt when SMPlayer was played (default conf without changing anything) was that the image seemed to be filtered like there was a certain effect, still on the same Mac and the same conf I played on MPV, the strange thing didn't happen and it seemed very natural and I I don't want to use equalizer software because it adds to the CPU load

**Comment by tonho911** [comment]
> It will probably work but the video will be on a separate window. I'm using a patched version to fix that problem.


error bro...

mpv --no-quiet --terminal --no-msg-color --input-ipc-server=/private/var/folders/k4/qwqyrpyx18799b76fsprrz6m0000gn/T/smplayer-mpv-459 --msg-level=ffmpeg/demuxer=error --video-rotate=no --no-config --no-fs --hwdec=no --sub-auto=fuzzy --shm-buffer-name=/smplayer-1113 --vo=shm, --no-keepaspect --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --sub-pos=100 --volume=55 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/Users/antoniech/Pictures/smplayer_screenshots --audio-pitch-correction=yes --volume-max=110 --term-playing-msg=MPV_VERSION=${=mpv-version:}
INFO_VIDEO_WIDTH=${=width}
INFO_VIDEO_HEIGHT=${=height}
INFO_VIDEO_ASPECT=${=video-params/aspect}
INFO_VIDEO_FPS=${=container-fps:${=fps}}
INFO_VIDEO_FORMAT=${=video-format}
INFO_VIDEO_CODEC=${=video-format}
INFO_DEMUX_ROTATION=${=track-list/0/demux-rotation}
INFO_AUDIO_FORMAT=${=audio-codec-name}
INFO_AUDIO_CODEC=${=audio-codec-name}
INFO_AUDIO_RATE=${=audio-params/samplerate}
INFO_AUDIO_NCH=${=audio-params/channel-count}
INFO_LENGTH=${=duration:${=length}}
INFO_DEMUXER=${=current-demuxer:${=demuxer}}
INFO_SEEKABLE=${=seekable}
INFO_TITLES=${=disc-titles}
INFO_CHAPTERS=${=chapters}
INFO_TRACKS_COUNT=${=track-list/count}
METADATA_TITLE=${metadata/by-key/title:}
METADATA_ARTIST=${metadata/by-key/artist:}
METADATA_ALBUM=${metadata/by-key/album:}
METADATA_GENRE=${metadata/by-key/genre:}
METADATA_DATE=${metadata/by-key/date:}
METADATA_TRACK=${metadata/by-key/track:}
METADATA_COPYRIGHT=${metadata/by-key/copyright:}
INFO_MEDIA_TITLE=${=media-title:}
INFO_STREAM_PATH=${stream-path}
 --audio-client-name=SMPlayer --term-status-msg=STATUS: ${=time-pos} / ${=duration:${=length:0}} P: ${=pause} B: ${=paused-for-cache} I: ${=core-idle} VB: ${=video-bitrate:0} AB: ${=audio-bitrate:0} /Users/antoniech/Desktop/4K VIDEO TESTER/Amazing Dolby Vision 4K HDR 120FPS.mkv

Error parsing option shm-buffer-name (option not found)
Setting commandline option --shm-buffer-name=/smplayer-1113 failed.
Exiting... (Fatal error)

**Reply by smplayer-dev** [reply]
Try setting the video output to GPU or libmpv (anything other than the default).

**Reply by tonho911** [reply]
the best output video in my opinion is libmpv but I don't know where the input.conf is as I usually work on MPV without SMPlayer, can you show me where the config is so I can apply it in the settings

**Reply by Raza-0007** [reply]
@tonho911 

Leave the output driver in smplayer preferences (`Preferences -> General -> Video`) to default.

Then pass this command `-vo=libmpv` in `Preferences -> Advanced -> Mplayer/mpv -> Options`

**Reply by tonho911** [reply]
Your method can work perfectly bro without editing the file ~/.config/smplayer/smplayer.ini and changing disable_player_config to false. But the problem is that the resulting aspect ratio is actually going to fullscreen which should be widescreen, where do you change it so that it returns to aspect widescreen MPV default ratio bro?

**Reply by Raza-0007** [reply]
I have responded in your other thread [here](https://github.com/smplayer-dev/smplayer/issues/875#issuecomment-1876210844)
